### PR TITLE
fix(electron): Clear events button fixes for electron app.

### PR DESCRIPTION
**Pull request by Shubhdeep12**
Before opening this PR:

- [x] I added a [Changeset Entry](https://spotlightjs.com/contribute/changesets/) with `pnpm changeset:add`
- [x] I referenced issues that this PR addresses

fixes: #350  

1. Removed clear events button for electron app.
2. Added clear events function in Reload and force reload toolbar options in electron app.
3. Created a event listener to clear events data.
4. Added a reset function in integration.
5. Updated the docs and added `reset` function in integration type and `showClearEventsButton` for configuration page

### CODE REVIEW

File: packages/overlay/src/components/Debugger.tsx [hunk 1]
```diff
@@ -57,30 +49,9 @@ export default function Debugger({
   isOnline: boolean;
   setTriggerButtonCount: (count: NotificationCount) => void;
   fullPage: boolean;
-  setReloadSpotlight: React.Dispatch<React.SetStateAction<number>>;
 }) {
   const isSentryIntegrationAdded = integrations.some(integration => integration.name === 'sentry');
-  const { setEvents } = useContext(SentryEventsContext);
```
Comment (HazAT) [inline packages/overlay/src/components/Debugger.tsx:63]: I would rather not have any traces of Electron app in Spotlight itself when not absolutely necessary. Rather let's add an option - `showClearEventsButton` to Spotlight itself, which can be default `true` and in the electron app when we init Spotlight we set it to false.

File: packages/overlay/src/App.tsx [hunk 2]
```diff
@@ -81,11 +108,13 @@ export default function App({
     spotlightEventTarget.addEventListener('open', onOpen as EventListener);
     spotlightEventTarget.addEventListener('close', onClose);
     spotlightEventTarget.addEventListener('navigate', onNavigate as EventListener);
+    spotlightEventTarget.addEventListener('sentry:clearEvents', clearEvents as EventListener);
```
Comment (HazAT) [inline packages/overlay/src/App.tsx:111]: Any specific reason why `sentry:clearEvents`? 
I would just call it `clearEvents`

File: packages/overlay/src/App.tsx [hunk 3]
```diff
@@ -57,6 +60,30 @@ export default function App({
...
+    const sidecarUrlObject: URL = new URL(sidecarUrl);
+    const host: string = sidecarUrlObject?.hostname;
+    const port: string = sidecarUrlObject?.port;
+    const clearEventsUrl: string = `http://${host}:${port}/clear`;
+    makeFetch(clearEventsUrl, {
+      method: 'DELETE',
+      mode: 'cors',
+    }).catch(err => {
+      console.error(
+        `Sentry SDK can't connect to Sidecar is it running? See: https://spotlightjs.com/sidecar/npx/`,
```
Comment (HazAT) [inline packages/overlay/src/App.tsx:74]:
```suggestion
        `Spotlight can't connect to Sidecar is it running? See: https://spotlightjs.com/sidecar/npx/`,
```

File: packages/overlay/src/App.tsx [hunk 4]
```diff
@@ -1,8 +1,11 @@
-import { useEffect, useState } from 'react';
+import { useContext, useEffect, useState } from 'react';
 import { useNavigate } from 'react-router-dom';
 import Debugger from './components/Debugger';
 import Trigger from './components/Trigger';
 import type { Integration, IntegrationData } from './integrations/integration';
+import sentryDataCache from './integrations/sentry/data/sentryDataCache';
```
Comment (HazAT) [inline packages/overlay/src/App.tsx:6]: That's not ideal, we should find another way to do this - we are adding something from the Sentry integration to the main skeleton app making Spotlight practically required to always use Sentry.

We should refactor this that Spotlight forwards all incoming events to all integrations. Integrations that also listen to the event do something then.

**Review by HazAT — CHANGES_REQUESTED** [review]

File: packages/overlay/src/components/Debugger.tsx (hunk 1 above)
Comment (Shubhdeep12) [reply packages/overlay/src/components/Debugger.tsx:63]: 👍🏻

File: packages/overlay/src/App.tsx (hunk 2 above)
Comment (Shubhdeep12) [reply packages/overlay/src/App.tsx:111]: Right now, it clears out the data from the sidecar and events stored for sentry integration only. so I thought to make it just for sentry integration
that was the reason to add sentry: here 
and to check if sentry integration was added, only show the clear events button.

but I made that generic now, so I am changing this to "clearEvents"

File: packages/overlay/src/App.tsx (hunk 3 above)
Comment (Shubhdeep12) [reply packages/overlay/src/App.tsx]: 👍🏻

File: packages/overlay/src/App.tsx (hunk 4 above)
Comment (Shubhdeep12) [reply packages/overlay/src/App.tsx]: Yes @HazAT , that makes sense.

In earlier commits, when we added clear events, I only implemented it for Sentry. However, this approach could potentially break the spotlight if the user hasn't added Sentry integration.

To address this, I've updated the logic by checking for the existence of the `processEvent` function for every integration. If the function is present, we clear the `integrationData` for that specific integration(which includes the events for every integration)
Additionally, for integrations like Sentry, which process events but have internal storage (like we have `sentryDataCache`), I've added a reset function in the integration. We will call this function while clearing events, allowing users to clear events in the integration using the button.

Approach Summary:
- Check for the existence of the processEvent function for each integration.
- If the function is present, clear the integration data (including events).
- For integrations with internal storage, implement a reset function in the integration.
- Call the reset function while clearing events to enable users to clear events using the button.

**Review by HazAT — APPROVED** [review]
🥇

**Pull request merged**